Sources/TidepoolKit/TAPI.swift

```diff
@@ -39,43 +36,57 @@ public class TAPI {
     /// The URLSessionConfiguration used for all requests. The default is typically acceptable for most purposes. Any changes
     /// will only apply to subsequent requests.
     public var urlSessionConfiguration: URLSessionConfiguration {
-        get {
-            return urlSessionConfigurationLocked.value
-        }
-        set {
-            urlSessionConfigurationLocked.mutate { $0 = newValue }
-            urlSessionLocked.mutate { $0 = nil }
+        didSet {
+            urlSession = URLSession(configuration: urlSessionConfiguration)
         }
     }
 
+    public func setURLSessionConfiguration(_ configuration: URLSessionConfiguration) {
+        urlSessionConfiguration = configuration
+    }
+
     /// The session used for all requests.
     public var session: TSession? {
-        get {
-            return sessionLocked.value
-        }
-        set {
-            sessionLocked.mutate { $0 = newValue }
-            observers.forEach { $0.apiDidUpdateSession(newValue) }
+        didSet {
+            observers.forEach { $0.apiDidUpdateSession(self.session) }
```

Comment:
just wanted to confirm that the removal of locked parameters is result of using `@MainActor`, right?

Reply:
Yes